### PR TITLE
Use temporary file to save bitmaps

### PR DESCRIPTION
**Pull request by eupn**
Removes `TODO`.

### CODE REVIEW

File: store/src/leaf_set.rs [hunk 1]
```diff
@@ -168,13 +168,26 @@ impl LeafSet {
 		// First run the optimization step on the bitmap.
 		self.bitmap.run_optimize();
 
-		// TODO - consider writing this to disk in a tmp file and then renaming?
-
 		// Write the updated bitmap file to disk.
 		{
-			let mut file = BufWriter::new(File::create(self.path.clone())?);
+			// Write temporary file
```
Comment (hashmap) [inline store/src/leaf_set.rs]: It seems like a common pattern, can we extract it into function which takes a closure or macros (if we have to)?

Comment (ignopeverell) [reply store/src/leaf_set.rs]: Given that you have it twice here already, I'd second @hashmap. It'll also have the nice consequence of making those 2 functions more readable, all the file manipulations hides a little bit what they're really doing.

Comment (eupn) [reply store/src/leaf_set.rs]: 👌

**Comment by eupn** [comment]
r=@ignopeverell 
r=@hashmap

**Pull request merged**